homeassistant/components/madvr/coordinator.py

```diff
@@ -0,0 +1,80 @@
+"""Coordinator for handling data fetching and updates."""
+
+import logging
+
+from madvr.madvr import Madvr
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MadVRCoordinator(DataUpdateCoordinator):
+    """My custom coordinator for push-based API."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        my_api: Madvr,
+        mac: str,
+        name: str,
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="My API Coordinator",
+        )
+        self.entry_id = config_entry.entry_id
+        self.my_api = my_api
+        self.mac = mac
+        self.name = name
+        self.previous_data: dict = {}
+        self.my_api.set_update_callback(self.handle_push_data)
+        _LOGGER.debug("MadVRCoordinator initialized")
+
+    async def _async_update_data(self):
+        """No-op method for initial setup."""
+        return self.previous_data
```

Comment:
euh?

Reply:
I had a HA error unless I added it but could have been caused by something else. I will try to remove